GiniSwitchSDK/GiniSwitchSDK/Classes/MultiPageCoordinator.swift

```diff
@@ -142,28 +143,33 @@ class MultiPageCoordinator {
         if extractionsCompleted {
             // reset the flag just in case
             extractionsCompleted = false
-            // show the extractions completed screen
-            let completionController = UIStoryboard.switchStoryboard()?.instantiateViewController(withIdentifier: "ExtractionsCompletedViewController") as! ExtractionsCompletedViewController
-            completionController.image = GiniSwitchAppearance.analyzedImage
-            completionController.text = GiniSwitchAppearance.analyzedText
-            completionController.textSize = GiniSwitchAppearance.analyzedTextSize
-            completionController.textColor = GiniSwitchAppearance.analyzedTextColor
-            multiPageViewController.present(controller: completionController, presentationStyle: .modal, animated: true) { [weak self] in
-                // after it is presented, push the extractions screen below it. That way, when it is
-                // automatically dismissed, the extractions will appear below
-                guard let weakSelf = self else {
-                    return
+            if shouldShowCompletionScreen {
+                // show the extractions completed screen
+                let completionController = UIStoryboard.switchStoryboard()?.instantiateViewController(withIdentifier: "ExtractionsCompletedViewController") as! ExtractionsCompletedViewController
+                completionController.image = GiniSwitchAppearance.analyzedImage
+                completionController.text = GiniSwitchAppearance.analyzedText
+                completionController.textSize = GiniSwitchAppearance.analyzedTextSize
+                completionController.textColor = GiniSwitchAppearance.analyzedTextColor
+                multiPageViewController.present(controller: completionController, presentationStyle: .modal, animated: true) { [weak self] in
+                    // after it is presented, push the extractions screen below it. That way, when it is
+                    // automatically dismissed, the extractions will appear below
+                    guard let weakSelf = self else {
```

Comment:
That is why I always prefer to use the trick of `'self' = self` rather than naming it as `weakSelf` or `strongSelf`, because for me if you unwrap it is not weak anymore but I understand that you refer it as `weak` since it was a `weak self` before unwrapping it. It is fine if you keep it.

Reply:
I actually didn't know about that. What I don't like about it, is having those quotes and using a reserved word like `self`as a variable name.

But I really like not having `weakSelf`. So lets try it!